src/vcpkg/configure-environment.cpp

```diff
@@ -10,57 +10,107 @@
 #include <vcpkg/vcpkgcmdarguments.h>
 #include <vcpkg/vcpkgpaths.h>
 
+#if defined(VCPKG_CE_SHA)
+#define VCPKG_CE_SHA_AS_STRING MACRO_TO_STRING(VCPKG_CE_SHA)
+#endif // ^^^ VCPKG_CE_SHA
+
+namespace
+{
+    using namespace vcpkg;
+    DECLARE_AND_REGISTER_MESSAGE(FailedToProvisionCe, (), "", "Failed to provision vcpkg-ce.");
+    DECLARE_AND_REGISTER_MESSAGE(VcpkgCeIsExperimental,
+                                 (),
+                                 "",
+                                 "vcpkg-ce ('configure environment') is experimental and may change at any time.");
+    DECLARE_AND_REGISTER_MESSAGE(DownloadingVcpkgCeBundle,
+                                 (msg::version),
+                                 "",
+                                 "Downloading vcpkg-ce bundle {version}...");
+    DECLARE_AND_REGISTER_MESSAGE(DownloadingVcpkgCeBundleLatest,
+                                 (),
+                                 "This message is normally displayed only in development.",
+                                 "Downloading latest vcpkg-ce bundle...");
+
+    void extract_ce_tarball(const VcpkgPaths& paths,
+                            const Path& ce_tarball,
+                            const Path& node_path,
+                            const Path& node_modules)
+    {
+        auto& fs = paths.get_filesystem();
+        fs.remove_all(node_modules, VCPKG_LINE_INFO);
+        Path node_root = node_path.parent_path();
+        auto npm_path = node_root / "node_modules" / "npm" / "bin" / "npm-cli.js";
+        if (!fs.exists(npm_path, VCPKG_LINE_INFO))
+        {
+            npm_path = Path(node_root.parent_path()) / "lib" / "node_modules" / "npm" / "bin" / "npm-cli.js";
+        }
+
+        Command cmd_provision(node_path);
+        cmd_provision.string_arg(npm_path);
+        cmd_provision.string_arg("--force");
+        cmd_provision.string_arg("install");
+        cmd_provision.string_arg("--no-save");
+        cmd_provision.string_arg("--no-lockfile");
+        cmd_provision.string_arg("--scripts-prepend-node-path=true");
+        cmd_provision.string_arg("--silent");
+        cmd_provision.string_arg(ce_tarball);
+        auto env = get_modified_clean_environment({}, node_root);
+        const auto provision_status = cmd_execute(cmd_provision, WorkingDirectory{paths.root}, env);
+        fs.remove(ce_tarball, VCPKG_LINE_INFO);
+        if (provision_status != 0)
+        {
+            fs.remove_all(node_modules, VCPKG_LINE_INFO);
+            Checks::msg_exit_with_error(VCPKG_LINE_INFO, msgFailedToProvisionCe);
+        }
+    }
+}
+
 namespace vcpkg
 {
     int run_configure_environment_command(const VcpkgPaths& paths, View<std::string> args)
     {
-        print2(Color::warning, "vcpkg-ce ('configure environment') is experimental and may change at any time.\n");
+        msg::println(Color::warning, msg::format(msg::msgWarningMessage).append(msg::format(msgVcpkgCeIsExperimental)));
```

Comment:
we may also want a `msg::print_warning` or smth?

Reply:
Added.